flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1748,3 +1934,180 @@ def get_mounts():
+        # [mount] = LoggedAction.of_type(logger.messages, MOUNT_VOLUME)
+        # self.assertEqual([unmount, detach, resize], action.children)
+        1/0
+
+    @validate_logging(verify_run_log)
+    def test_run_grow(self, logger):
+        """
+        After running ``ResizeBlockDeviceDataset``, its volume has been
+        resized.
+
+        XXX: We manually format and mount the created volume to avoid also
+        exercising the ``CreateBlockDeviceDataset`` state change in the same
+        unit test.
+        """
+        # self.patch(blockdevice, "_logger", logger)
+
+        # node = u"192.0.2.3"
+        # dataset_id = uuid4()
+        # api = loopbackblockdeviceapi_for_test(self)
+        # volume = api.create_volume(
+        #     dataset_id=dataset_id, size=REALISTIC_BLOCKDEVICE_SIZE
+        # )
+        # volume = api.attach_volume(volume.blockdevice_id, node)
+        # device = api.get_device_path(volume.blockdevice_id)
+        # mountroot = mountroot_for_test(self)
+        # mountpoint = mountroot.child(unicode(dataset_id).encode("ascii"))
+        # mountpoint.makedirs()
+        # check_output([b"mkfs", b"-t", b"ext4", device.path])
+        # check_output([b"mount", device.path, mountpoint.path])
+
+        # deployer = BlockDeviceDeployer(
+        #     hostname=node,
+        #     block_device_api=api,
+        #     mountroot=mountroot,
+        # )
+        # change = ResizeBlockDeviceDataset(volume=volume, size=REALISTIC_BLOCKDEVICE_SIZE * 2)
+        # self.successResultOf(change.run(deployer))
+
+        # expected_volume = volume.set(size=REALISTIC_BLOCKDEVICE_SIZE * 2)
+
+        # self.assertEqual([expected_volume], api.list_volumes())
+        1/0
+
+    def test_run_shrink(self, logger):
+        """
+        After running ``ResizeBlockDeviceDataset``, its volume has been
+        shrunk.
+
+        XXX: This can be marked TODO and implemented later in FLOC-1503.
+        """
+        1/0
+
+    def test_resize_unmounted_volume(self):
+        """
+        ``ResizeBlockDeviceDataset`` can resize an attached but unmounted volume.
+
+        The result is a larger block device with an expanded filesystem
+        containing the data that was their before the resize.
+        """
+        # XXX This situation shouldn't be possible, right?
+        1/0
+
+    def test_resize_unformatted_volume(self):
+        """
+        ``ResizeBlockDeviceDataset`` can resize an unformatted volume.
+
+        The result is a larger block device which is attached to the host but
+        which does not yet have a filesystem.
+        """
+        # XXX This situation shouldn't be possible, right?
+        1/0
+
+    def test_resize_unattached_volume(self):
+        """
+        ``ResizeBlockDeviceDataset`` can resize an unattached, unformatted volume.
+
+        The result is a larger block device which is unattached and without filesystem.
+        """
+        # XXX This situation shouldn't be possible, right?
```

Comment:
Hmmm.  As long as it's not attached maybe it doesn't matter if it's the wrong size.  As soon as someone actually wants to use it they'll have to attach it and then it'll get resized.  Probably good enough for now, at least.